index.ts

```diff
@@ -93,4 +106,18 @@ if (conf.get('websocket.ws.enable')) {
     });
     var wss = new webSocket.Server({ server: serverWS.server });
     wss.on('connection', webSocketHandler.wsOnConnect);
+    pServers.push(new Promise<void>((resolve: () => void,
+                                     reject: (error: Error) => void): void => {
+        serverWS.on('listening', (): void => {
+            resolve();
+        });
+    }));
```

Comment:
Very similar code is repeated 3 times with the only difference being the restify server variable.
I believe there is an opportunity here for refactoring into a function.